Lib/test/test_hashlib.py

```diff
@@ -371,6 +373,32 @@ def check(self, name, data, hexdigest, shake=False, **kwargs):
             if not shake:
                 self.assertEqual(len(digest), m.digest_size)
 
+        if not shake and kwargs.get("key") is None:
+            # skip shake and blake2 extended parameter tests
+            self.check_file_digest(name, data, hexdigest)
+
+    def check_file_digest(self, name, data, hexdigest):
+        hexdigest = hexdigest.lower()
+        digests = [name]
+        digests.extend(self.constructors_to_test[name])
+
+        for digest in digests:
+            with self.subTest(digest=digest):
+                buf = io.BytesIO(data)
+                buf.seek(0)
+                self.assertEqual(
+                    hashlib.file_digest(buf, digest).hexdigest(), hexdigest
+                )
+                with open(os_helper.TESTFN, "wb") as f:
```

Comment:
create a single data file outside of the digest subTest loop.  only unlink it afterwards, perhaps via os.addCleanup rather than using try/finally.

Reply:
Some tests call the ``check`` function multiple times, so addCleanup would try to remove the file multiple times. I wrapped block into a try-finally.